snuba/admin/views.py

```diff
@@ -441,8 +442,65 @@ def clickhouse_trace_query() -> Response:
         )
 
     try:
-        result = run_query_and_get_trace(storage, raw_sql)
-        return make_response(jsonify(asdict(result)), 200)
+        # Append 'log_profile_events=1' with/without the SETTINGS clause to the incoming query.
+        settings_index = raw_sql.lower().find("settings")
+        if settings_index != -1:
+            start_index = settings_index + len("settings")
+            remaining = raw_sql[start_index:].strip()
+            if "log_profile_events=1" not in remaining:
+                raw_sql += ", log_profile_events=1"
+        else:
+            raw_sql += " SETTINGS log_profile_events=1"
+
+        query_trace = run_query_and_get_trace(storage, raw_sql)
+
+        profile_events_raw_sql = "SELECT ProfileEvents FROM system.query_log WHERE query_id = '{}' AND type = 'QueryFinish'"
+        hosts_ports_query_ids_nodes = parse_trace_for_query_ids(query_trace, storage)
+
+        for host_port_query_id_node in hosts_ports_query_ids_nodes:
+            sql = profile_events_raw_sql.format(host_port_query_id_node["query_id"])
+            system_query_result, counter = None, 0
+
+            while counter < 10:
+                # There is a race between the original query to trace finishing and the system query log populating.
+                # Sleep between the query executions.
+                system_query_result = run_system_query_on_host_with_sql(
+                    host_port_query_id_node["host"],
+                    int(host_port_query_id_node["port"]),
+                    storage,
+                    sql,
+                )
+                if not system_query_result.results:
+                    time.sleep(1)
+                    counter += 1
+                else:
+                    break
+
+            if system_query_result is None or len(system_query_result.results) == 0:
+                raise ClickHouseTimeoutError(
+                    "Timed out while executing: {}".format(sql)
```

Comment:
nit: I think this error message could be misleading. ClickHouse is serving these queries without timing out; we're just waiting for a table to get filled and that isn't happening (probably because query log isn't enabled or something). What you're timing out for is waiting for a query log row to be inserted.

I call this a nit because nothing else uses your `ClickHouseTimeoutError` class but it would be more important if that thing had wider use.